pkg/chunk/purger/purger.go

```diff
@@ -248,6 +248,10 @@ func (p *Purger) workerJobCleanup(job workerJob) {
 			default:
 				// already sent
 			}
+		} else if len(p.usersWithPendingRequests) == 0 {
```

Comment:
[open question, just to understand] I'm wondering if this `else if` should actually be an `if {`. What if the previous `if` branch deletes the latest `p.usersWithPendingRequests`?

Reply:
If the previous `if {` is true then that means there are more pending requests and the age of oldest pending request is non-zone. Even if I set it to 0 by changing this `else if` to an independent `if` condition then it would still get changed to a non-zero value when the requests get loaded. I don't see much of a difference except a dip to 0 if samples get scrapped before pending requests are found during the next load of requests.